CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 # Copyright (c) m8mble 2020.
 # SPDX-License-Identifier: BSL-1.0
 
-cmake_minimum_required(VERSION 3.17)
+cmake_minimum_required(VERSION 3.21)
```

Comment:
Thanks for spotting this. Let me suggest
```cmake
set(CLEANTEST_IS_TOP_LEVEL OFF)
if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
    set(CLEANTEST_IS_TOP_LEVEL ON)
endif()
```
(and replacing `PROJECT_IS_TOP_LEVEL => CLEANTEST_IS_TOP_LEVEL`) as a workaround in order to stay with the `cmake` version.